src/Core/functions.php

```diff
@@ -272,12 +272,13 @@ function triggerWarning(string $message): void
     /**
      * Helper method for outputting deprecation warnings
      *
+     * @param string $version The version that added this deprecation warning.
      * @param string $message The message to output as a deprecation warning.
      * @param int $stackFrame The stack frame to include in the error. Defaults to 1
      *   as that should point to application/plugin code.
      * @return void
      */
-    function deprecationWarning(string $message, int $stackFrame = 1): void
+    function deprecationWarning(string $version, string $message, int $stackFrame = 1): void
```

Comment:
Putting the version first is going to make 4.next merging harder. Should we have the parameters be `message, version`?

Reply:
If you think that will help. The merge will need a parameter added either way.

Reply:
True. Lets leave it this way then. The deprecation messages will be obviously wrong 😛

Reply:
You can also add an assert() for the x.y.z format :) If you want to make double sure.

Reply:
Well, it will throw an error on a missing parameter or a TypeError because an int was passed in.